File: docs/current/cookbook/snippets/modules-shared-client/functions/alpine/alpine.go
package alpine

import (
	"context"

	"dagger.io/dagger"
)

// create base image
func base(client *dagger.Client) *dagger.Container {
	return client.
		Container().
		From("alpine:latest")
}

// run command in base image
func Version(client *dagger.Client) string {
	ctx := context.Background()

	out, err := base(client).
		WithExec([]string{"cat", "/etc/alpine-release"}).
		Stdout(ctx)
	if err != nil {
		panic(err)
	}

	return out
}

File: docs/current_docs/cookbook/snippets/modules-shared-client/classes/alpine.py
import dagger


class Alpine:
    __client: dagger.Client

    # initialize pipeline class
    def __init__(self, client: dagger.Client):
        self.__client = client

    # get base image
    def __base(self):
        return self.__client.container().from_("alpine:latest")

    # run command in base image
    async def version(self):
        return await self.__base().with_exec(["cat", "/etc/alpine-release"]).stdout()
